packages/kbn-alerts-ui-shared/src/rule_type_modal/components/rule_type_list.tsx

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React from 'react';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFacetGroup,
+  EuiFacetButton,
+  EuiCard,
+  EuiSpacer,
+  EuiText,
+  EuiEmptyPrompt,
+  EuiButton,
+} from '@elastic/eui';
+import { omit } from 'lodash';
+import { PRODUCER_DISPLAY_NAMES } from '../../common/i18n';
+import { RuleTypeWithDescription, RuleTypeCountsByProducer } from '../types';
+
+interface RuleTypeListProps {
+  ruleTypes: RuleTypeWithDescription[];
+  onSelectRuleType: (ruleTypeId: string) => void;
+  onFilterByProducer: (producer: string | null) => void;
+  selectedProducer: string | null;
+  ruleTypeCountsByProducer: RuleTypeCountsByProducer;
+  onClearFilters: () => void;
+}
+
+const producerToDisplayName = (producer: string) => {
+  return Reflect.get(PRODUCER_DISPLAY_NAMES, producer) ?? producer;
+};
+
+export const RuleTypeList: React.FC<RuleTypeListProps> = ({
+  ruleTypes,
+  onSelectRuleType,
+  onFilterByProducer,
+  selectedProducer,
+  ruleTypeCountsByProducer,
+  onClearFilters,
+}) => {
+  const rulesList = [...ruleTypes].sort((a, b) => a.name.localeCompare(b.name));
+
+  return (
+    <EuiFlexGroup
+      style={{
+        height: '100%',
+      }}
+    >
+      <EuiFlexItem
+        grow={1}
+        style={{ paddingTop: 16 /* Match drop shadow padding in the right column */ }}
```

Comment:
nitpick: for static styles, let's move them out of the component so they don't cause re-renders

Reply:
Also preferable to use `useEuiPaddingCSS` to ensure our padding is consistent: https://eui.elastic.co/#/theming/sizing/utilities#padding

Reply:
I can move this out of the component, or I can use a hook, but I can't do both

Reply:
Moving them out of the component causes typescript issues so I guess I'll keep them in and memoize

Reply:
good idea!

Reply:
I was able to do some research and a quick test and determined that memoizing `style` props have no effect on React re-renders. React seems to be doing deep equality comparison on `style`.